python/ray/serve/controller.py

```diff
@@ -583,6 +583,11 @@ async def get_serve_status(self) -> bytes:
                 except RayTaskError:
                     serve_app_status = ApplicationStatus.DEPLOY_FAILED
                     serve_app_message = f"Deployment failed:\n{traceback.format_exc()}"
+                except RuntimeEnvSetupError:
+                    serve_app_status = ApplicationStatus.DEPLOY_FAILED
+                    serve_app_message = (
+                        f"Runtime env setup failed:\n{traceback.format_exc()}"
+                    )
 
```

Comment:
I would suggest we have a generic `except Exception` block here that also makes it `DEPLOY_FAILED` with a message something along the lines of "unknown error occurred: {exception_message}"